### ISSUE TITLE
Resolver bound using an interface fail to resolve at runtime

### ISSUE THREAD

**Issue by FolkCoder** [ChilliCream/hotchocolate#2742]
Resolvers bound to the request executor that use an interface rather than a concrete class cannot be resolved at runtime, throwing the following exception:

```json
      "extensions": {
        "message": "The type HotChocolateDependencyResolutionBug.Resolvers.ITestClassResolver is abstract and we cannot use it to compile a service factory from it.",
        "stackTrace": "   at HotChocolate.Utilities.ActivatorHelper.ResolveConstructor(Type type)\r\n   at HotChocolate.Utilities.ActivatorHelper.CreateNewInstance(Type type, ParameterExpression services)\r\n   at HotChocolate.Utilities.ActivatorHelper.<>c__DisplayClass4_0.<CompileFactory>b__0(Type _)\r\n   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)\r\n   at HotChocolate.Utilities.ActivatorHelper.CompileFactory(Type type)\r\n   at HotChocolate.Execution.Processing.DefaultActivator.CreateInstance(Type type)\r\n   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)\r\n   at HotChocolate.Execution.Processing.DefaultActivator.GetOrCreate(Type type)\r\n   at HotChocolate.Execution.Processing.DefaultActivator.GetOrCreate[T]()\r\n   at HotChocolate.Execution.Processing.MiddlewareContext.Resolver[T]()\r\n   at lambda_method20(Closure , IResolverContext )\r\n   at HotChocolate.Types.FieldMiddlewareCompiler.<>c__DisplayClass3_0.<<CreateResolverMiddleware>b__0>d.MoveNext()\r\n--- End of stack trace from previous location ---\r\n   at HotChocolate.Execution.Processing.ResolverTask.ExecuteResolverPipelineAsync(CancellationToken cancellationToken)\r\n   at HotChocolate.Execution.Processing.ResolverTask.TryExecuteAsync(CancellationToken cancellationToken)"
      }
```

This seems to be changed behavior from v10, which successfully resolved resolvers from the DI container rather than requiring a concrete class to be provided to the schema builder.

A minimal solution demonstrating the behavior can be found [here](https://github.com/FolkCoder/HotChocolateDependencyResolutionBug).

**Comment by michaelstaib** [comment]
With the June release we are doing a big push for schema first and then this should be fixed.

**Comment by apitzele** [comment]
this has been resolved (at least as of 11.1).  I have no issues binding schema first resolvers to interfaces.
